validation/generate_test.go

```diff
@@ -0,0 +1,57 @@
+package validation
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	rfc2119 "github.com/opencontainers/runtime-tools/error"
+	"github.com/opencontainers/runtime-tools/generate"
+	"github.com/opencontainers/runtime-tools/specerror"
+	"github.com/opencontainers/runtime-tools/validate"
+)
+
+// Smoke test to ensure that _at the very least_ our default configuration
+// passes the validation tests. If this test fails, something is _very_ wrong
+// and needs to be fixed immediately (as it will break downstreams that depend
+// on us for a "sane default" and do compliance testing -- such as umoci).
+func TestGenerateValid(t *testing.T) {
+	bundle, err := ioutil.TempDir("", "TestGenerateValid_bundle")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(bundle)
+
+	// Create our toy bundle.
+	rootfsPath := filepath.Join(bundle, "rootfs")
+	if err := os.Mkdir(rootfsPath, 0755); err != nil {
+		t.Fatal(err)
+	}
+	configPath := filepath.Join(bundle, "config.json")
+	g := generate.New()
+	if err := (&g).SaveToFile(configPath, generate.ExportOptions{Seccomp: false}); err != nil {
+		t.Fatal(err)
+	}
+
+	// Validate the bundle.
+	v, err := validate.NewValidatorFromPath(bundle, true, runtime.GOOS)
+	if err != nil {
+		t.Errorf("unexpected NewValidatorFromPath error: %+v", err)
+	}
+	if err := v.CheckAll(); err != nil {
+		levelErrors, err2 := specerror.SplitLevel(err, rfc2119.Must)
+		if err2 != nil {
+			t.Errorf("unexpected error(s) filtering errors default: %+v", err2)
+			t.Errorf("unexpected error(s): %+v", err)
```

Comment:
What you call `err2` will only ever be non-`nil` if [it exactly matches the input `error`][1].  So I think you want to drop the previous line and keep only this one to avoid redundancy.  Perhaps the `SplitLevel` comment can also be improved to make this more clear?

[1]: https://github.com/opencontainers/runtime-tools/commit/929bd5e5ec3823e41c8c8cd362d30ea9645d63e1#diff-87c0b1eb29ae71890bc6a05f207759edR188